src/Symfony/Component/Security/Csrf/TokenStorage/CookieTokenStorageListener.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Csrf\TokenStorage;
+
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+
+/**
+ * Checks the request's attributes for a CookieTokenStorage instance. If one is
+ * found, the cookies representing the storage's changeset are appended to the
+ * response headers.
+ *
+ * TODO where to put this class?
```

Comment:
Was speculating about, but then i saw that every other kernel listener sits in the EventListener namespace of the HttpKernel component.